src/frontend/components/DashboardVideoPane/index.tsx

```diff
@@ -0,0 +1,180 @@
+import { Box } from 'grommet';
+import React, { useEffect, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router';
+import { Dispatch } from 'redux';
+import styled from 'styled-components';
+
+import { addResource } from '../../data/genericReducers/resourceById/actions';
+import { RootState } from '../../data/rootReducer';
+import { API_ENDPOINT } from '../../settings';
+import { appStateSuccess } from '../../types/AppData';
+import { modelName } from '../../types/models';
+import { uploadState, Video } from '../../types/tracks';
+import { report } from '../../utils/errors/report';
+import { DashboardInternalHeading } from '../Dashboard/DashboardInternalHeading';
+import { DashboardThumbnail } from '../DashboardThumbnail';
+import { DashboardVideoPaneButtons } from '../DashboardVideoPaneButtons/DashboardVideoPaneButtons';
+import { DashboardVideoPaneDownloadOptionConnected } from '../DashboardVideoPaneDownloadOptionConnected/DashboardVideoPaneDownloadOptionConnected';
+import { DashboardVideoPaneHelptext } from '../DashboardVideoPaneHelptext/DashboardVideoPaneHelptext';
+import { DashboardVideoPaneProgressConnected } from '../DashboardVideoPaneProgressConnected/DashboardVideoPaneProgressConnected';
+import { ERROR_COMPONENT_ROUTE } from '../ErrorComponent/route';
+import { UploadStatusPicker } from '../UploadStatusPicker/UploadStatusPicker';
+
+const { ERROR, PENDING, PROCESSING, READY, UPLOADING } = uploadState;
+
+const messages = defineMessages({
+  title: {
+    defaultMessage: 'Video status',
+    description:
+      'Subtitle for the video part of the dashboard (right now the only part until we add timed text tracks).',
+    id: 'components.DashboardVideoPane.title',
+  },
+});
+
+const DashboardVideoPaneInnerContainer = styled.div`
+  flex-grow: 1;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-around;
+  padding: 1rem;
+`;
+
+const DashboardVideoPaneInternalHeading = styled(DashboardInternalHeading)`
+  padding: 0 1rem 0 0;
+`;
+
+/** Props shape for the DashboardVideoPane component. */
+interface BaseDashboardVideoPaneProps {
+  jwt: string;
+  updateVideo: (video: Video) => void;
+  video: Video;
+}
+
+const BaseDashboardVideoPane = ({
+  jwt,
+  updateVideo,
+  video,
+}: BaseDashboardVideoPaneProps) => {
+  const [error, setError] = useState(false);
+  const [pollInterval, setPollInterval] = useState();
+
+  const pollForVideo = async () => {
```

Comment:
I think at the beginning of this function we should test if the current is ready. If yes we don't poll it, it's useless.

Reply:
IMO it made more sense to put the check in `useEffect` as a condition to start polling

I just did this 👍